buildSrc/src/main/groovy/org/gradle/binarycompatibility/transforms/FindGradleJar.groovy

```diff
@@ -32,8 +32,8 @@ class FindGradleJar extends ArtifactTransform {
 
     @Override
     List<File> transform(final File file) {
-        if (file.name.startsWith("gradle-${target}-")) {
-            [file]
+        if (file.name == 'gradle-jars') {
+            file.listFiles().findAll { it.name.startsWith("gradle-${target}-") } as List<File>
```

Comment:
Don't we need to sort that files to get the same classpath (order is important)?